### DISCUSSION TITLE
sgx_qv_verify_quote failed: 57372

### DISCUSSION THREAD

**Discussion by honorifi** [gramineproject/gramine#1537] [General]
I try to run the RA-TLS sample under directory 'CI-Examples/ra-tls-mbedtls'
but I got error after starting the client.

here's my commands:
```shell
make clean
make app dcap RA_TYPE=dcap
gramine-sgx ./server &
./client dcap \
    <MRENCLAVE of the server enclave> \
    <MRSIGNER of the server enclave> \
    <ISV_PROD_ID of the server enclave> \
    <ISV_SVN of the server enclave>
```
I also tried setting the measurement to zero to ignore them, but still got the same error:
```shell
make clean
make app dcap RA_TYPE=dcap
gramine-sgx ./server &
./client dcap 0 0 0 0
```

here's the error of the client:
<img width="846" alt="image" src="https://github.com/gramineproject/gramine/assets/49511267/12cc80ec-b898-4f65-b72e-e1cbaa947703">

here's the error of the server:
<img width="843" alt="image" src="https://github.com/gramineproject/gramine/assets/49511267/b1bb1ecd-49c9-4640-9976-9b093935f7c5">

I can't found solution on google, can anyone help?

**Comment by dimakuv** [comment]
The error returned by the Quote Verification library is 0xe01c.

https://github.com/intel/SGXDataCenterAttestationPrimitives/blob/6882afad8644c27db162b40994402c8ad2a7fb32/QuoteGeneration/quote_wrapper/common/inc/sgx_ql_lib_common.h#L79 -- it is `SGX_QL_QUOTE_CERTIFICATION_DATA_UNSUPPORTED`.

According to https://download.01.org/intel-sgx/latest/dcap-latest/linux/docs/Intel_SGX_ECDSA_QuoteLibReference_DCAP_API.pdf, it means:

> The certification type is not supported in the quote, which means PCK certificate chain doesn’t exist or not correct.

Which leads me to believe that you didn't install the PCCS infrastructure on your machine/cluster. (Alternatively, you could run these SGX examples on e.g. Microsoft Azure cloud which already installs this infrastructure for you.)

**Reply by honorifi** [reply]
Thanks a lot, I believe so. Afterwards, I tried installing libsgx-dcap-default-qpl following https://download.01.org/intel-sgx/sgx-dcap/1.14/linux/docs/Intel_SGX_SW_Installation_Guide_for_Linux.pdf, and this time I could not run server, it returns: 
![image](https://github.com/gramineproject/gramine/assets/49511267/c0a506fa-f9ec-4b13-b7ec-a71f83fa2960)
I believe this is because the pccs or aesm not configured correctly.
I still fail to run this example on my computer, is it quite difficult to deploy pccs on local machine? I'll try it on Azure cloud later if possible.
